### ISSUE TITLE
Arabic Text is breaking when using vector tile.

### ISSUE THREAD

**Issue by OwaisNadeemCentric** [Esri/esri-leaflet-vector#203]
### Describe the bug

When I try to use arcgis method to add a vector tile layer the arabic text doesn't break as you can see in the SS:

![image](https://github.com/Esri/esri-leaflet-vector/assets/122894004/88a969d7-d3b6-49c1-841a-894ddb67c917)

But when using esriVector the arabic text breaks.

![image](https://github.com/Esri/esri-leaflet-vector/assets/122894004/65c70179-4059-44b6-bd66-d274a08b0316)

`<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8" />
    <title>Esri Leaflet Vector Basemap</title>
    <meta name="viewport" content="width=device-width, initial-scale=1" />

    
    <link rel="stylesheet" href="https://unpkg.com/leaflet/dist/leaflet.css" />
    <script src="https://unpkg.com/leaflet/dist/leaflet.js"></script>

    
    <script src="https://unpkg.com/esri-leaflet/dist/esri-leaflet.js"></script>
    <script src="https://unpkg.com/esri-leaflet-vector@3/dist/esri-leaflet-vector.js"></script>
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/mapbox-gl/2.15.0/mapbox-gl.min.css" integrity="sha512-fPYzCDCwWGaqR93cxlCAZSqnHnabV5//RllFHLzi6L620mq7YtlyrBIUZbwttOkFFQgwVff/QMKpBPcQ4kH2dg==" crossorigin="anonymous" referrerpolicy="no-referrer" />
    <style>
      body {
        margin: 0;
        padding: 0;
      }

      #map {
        position: absolute;
        top: 0;
        bottom: 0;
        right: 0;
        left: 0;
      }
    </style>
  </head>

  <body>
    <div id="map"></div>
    <script>
      var map = L.map("map").setView([23.8859, 45.0792], 6);
      const tileApiKey = "xyz";
      L.esri.Vector.vectorTileLayer("xyz/Public_Basemap/VectorTileServer",{
      apikey: tileApiKey,
      }).addTo(map);
    </script>
  </body>

</html>`

### Reproduction

if you requires tileserver url and apiKey let me know. But it is confidential.

### System Info

```shell
latest version
```

**Comment by gavinr-maps** [comment]
Hi @OwaisNadeemCentric, thank you for the report.

We do have a sample for using other languages - with the Esri Basemaps service v2 - in the examples folder:
https://github.com/Esri/esri-leaflet-vector/blob/master/examples/languages.html

That looks correct to me (although I do not speak the language so it's hard for me to confirm):
![image](https://github.com/Esri/esri-leaflet-vector/assets/137905994/3be0d4e9-ca04-4778-a0c6-75de4485cba5)

But that sample uses `L.esri.Vector.vectorBasemapLayer`, where your example is using a custom Vector Tile layer using `L.esri.Vector.vectorTileLayer`, so it's a different use case. 


> if you requires tileserver url and apiKey let me know. But it is confidential.

In order to help out more, yes we would need a replication case. You could share us the URL and apiKey, but a better way would be if you can share the steps for us to create and publish a vector tile layer that demonstrates the issue. Or if you can share a publicly accessible layer that demonstrates the issue.

Thanks!

**Comment by OwaisNadeemCentric** [comment]
Thank you for looking at this issue I am sharing the current implementation with **esri-vector**

`<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />

    
    <script src="https://unpkg.com/leaflet/dist/leaflet.js"></script>

    
    <script src="https://unpkg.com/esri-leaflet/dist/esri-leaflet.js"></script>
    <script src="https://unpkg.com/esri-leaflet-vector@3/dist/esri-leaflet-vector.js"></script>
    <link rel="stylesheet" href="https://unpkg.com/leaflet/dist/leaflet.css" />

    <style>
      body {
        margin: 0;
        padding: 0;
      }

      #map {
        position: absolute;
        top: 0;
        bottom: 0;
        right: 0;
        left: 0;
      }
    </style>
  </head>

  <body>
    <div id="map"></div>
    
    <script>
      let map = L.map("map").setView([23.8859, 45.0792], 6);
      const apiKey =
        "lsDMaBfvbRC-tBrGd7LUCxoRxVbhH16ht_50KzUKPJNhqxwhCjbeDRWg3dziXOXTwCSV7VUdqURbs1pKqHCUFhKD9UZNnaFavcAgFdhJlBMFxNGgyugTAh8HQLy04Wu83uFQ4yCOZG4ebqR6_ygs22Xny-L5CW5KcHYjgGBg4veD7pbVk9aBwlFNXPrxic0e";

      const vectorTileUrl =
        "https://stgisudp.momra.gov.sa/server/rest/services/Hosted/Public_Basemap/VectorTileServer";
      const layer = new L.esri.Vector.vectorTileLayer(vectorTileUrl, {
        apikey: apiKey,
      });

      layer.addTo(map);
    </script>
  </body>
</html>
`

**Comment by gavinr-maps** [comment]
Thank you. Replication case:

- Using the ArcGIS Maps SDK for JavaScript,the labels are correctly on two lines: https://jsbin.com/zuronin/1/edit?html,output
- Using Esri Leaflet Vector, the labels are not on two lines: https://jsbin.com/ramerux/edit?html,output
  - Expected: labels on two lines for both
  - Actual: labels are not on two lines for the second example

**Comment by OwaisNadeemCentric** [comment]
Also I am sharing arcgis implementation try running it with live server.
[arcgis_map.zip](https://github.com/Esri/esri-leaflet-vector/files/12883527/arcgis_map.zip)

**Comment by OwaisNadeemCentric** [comment]
> Thank you. Replication case:
> 
> * Using the ArcGIS Maps SDK for JavaScript,the labels are correctly on two lines: https://jsbin.com/zuronin/1/edit?html,output
> * Using Esri Leaflet Vector, the labels are not on two lines: https://jsbin.com/ramerux/edit?html,output
>   
>   * Expected: labels on two lines for both
>   * Actual: labels are not on two lines for the second example

Arabic text is breaking also someone who can read arabic  can notice that actually arabic is starting from left instead of right

**Comment by gavinr-maps** [comment]
> also someone who can read arabic can notice that actually arabic is starting from left instead of right

thank you for pointing that out! I do not read Arabic so that was not obvious to me.

**Comment by OwaisNadeemCentric** [comment]
I have just noticed in your screenshot arabic is still breaking 

![image](https://github.com/Esri/esri-leaflet-vector/assets/122894004/2d04c5e0-f583-4818-99cb-fa00b6cc5ebf)

For better clarity as you can see in this image the arabic is broken as well as starting from left  : 
![image](https://github.com/Esri/esri-leaflet-vector/assets/122894004/fa57a7bf-34eb-48a5-8116-44fbe92cf9be)

and here you can see the letters are joining correctly as well as the first letter starts from right and so on.
![image](https://github.com/Esri/esri-leaflet-vector/assets/122894004/083f909a-852b-4047-a5b2-173b27740263)

**Comment by gavinr-maps** [comment]
> I have just noticed in your screenshot arabic is still breaking

Can you please confirm: the Arabic in [this screenshot](https://user-images.githubusercontent.com/122894004/274632443-2d04c5e0-f583-4818-99cb-fa00b6cc5ebf.png) is left-to-right when it should be right-to-left?

**Comment by OwaisNadeemCentric** [comment]
> > I have just noticed in your screenshot arabic is still breaking
> 
> Can you please confirm: the Arabic in [this screenshot](https://user-images.githubusercontent.com/122894004/274632443-2d04c5e0-f583-4818-99cb-fa00b6cc5ebf.png) is left-to-right when it should be right-to-left?

Yes In all of the screenshot arabic is starting from Left to Right.

**Comment by gavinr-maps** [comment]
Thank you for confirming that. Seems like we have two separate issues going on here:

1. The arabic is not properly left-to-right. 
    - As you may know, Esri Leaflet Vector uses the [Maplibre GL JS project](https://github.com/maplibre/maplibre-gl-js) for rendering the vector tile content. It seems like to get proper RTL support with Maplibre GL JS, you must use the "RTL plugin":
        - https://github.com/maplibre/maplibre-gl-js/issues/220#issuecomment-1048856008
        - https://github.com/mapbox/mapbox-gl-rtl-text
        - Our sample in https://github.com/Esri/esri-leaflet-vector/blob/master/examples/languages.html should properly use that plugin if it's necessary. Also we should have documentation that mentions this. I think we should move this to a separate issue

2. The label is not properly on a second line in Esri Leaflet Vector (https://github.com/Esri/esri-leaflet-vector/issues/203#issuecomment-1759694750)
    - After consulting with a colleague, my hypothesis here is that Esri Leaflet Vector must not be correctly applying the default line length of 10 ems (https://docs.mapbox.com/style-spec/reference/layers/#layout-symbol-text-max-width). 
    - This one may also be an issue with Mapbox GL JS. So a good next step might be to build a replication case with this layer using Mapbox GL JS to see if the issue happens there too.
    - One thing to try is if you can apply a line length rule to that layer in your vector tile style. If you could please try that and let us know if that resolves the issue or not.

**Comment by OwaisNadeemCentric** [comment]
Edit: see my separate issue here: https://github.com/Esri/esri-leaflet-vector/issues/204

So We can use mapbox-gl-rtl-text-plugin with esri to solve this issue ?

**Comment by gavinr-maps** [comment]
@OwaisNadeemCentric please move all discussion around the backwards RTL text to #204. Thanks
